### PR TITLE
initialization of multiple select content type

### LINKED ISSUE #424: Load data for select FieldType using a Symfony service
<a href="https://github.com/wachterjohannes"></a>
By @wachterjohannes _from_ https://github.com/sulu-cmf/SuluContentBundle/issues/222

Configurable over:
- service with interface (called from twig with webspace and locale)
- nesting params (property params)

### LINKED ISSUE #1740: Checkbox type caption
Add a caption to checkbox type and change style to toggler.

### LINKED ISSUE #1741: Checkbox group content type
implement a content type which has multiple checkboxes or radios. the values should be configurable over the params or a service (for 3rd-party webservices or xml loading).

### PR DESCRIPTION
**Pull request by danrot**
Requires https://github.com/massiveart/husky/pull/578 and https://github.com/massiveart/husky-validation/pull/53

__tasks:__

- [ ] test coverage
- [ ] submit changes to the documentation
- [x] Add behat tests for multiple and single select
- [x] Check block with zero elements
- [x] Handle old checkbox type
- [x] Fix editing of contact addresses

__informations:__

| q                | a
| ---------------- | ---
| Fixed tickets    | fixes #1741, fixes parts of #424, fixes #1740 
| BC breaks        | none
| Documentation PR | https://github.com/sulu-io/sulu-docs/pull/148

### CODE REVIEW

**Comment by wachterjohannes** [comment]
missing changelog

**Comment by wachterjohannes** [comment]
please update also the example template and run there the behat tests to be sure that they are running under travis.

File: src/Sulu/Bundle/ContentBundle/Content/Types/MultipleSelect.php [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+ */
+
+namespace Sulu\Bundle\ContentBundle\Content\Types;
+
+use Sulu\Component\Content\SimpleContentType;
+
+/**
+ * ContentType for a multiple select. Currently only support for checkboxes.
+ */
+class MultipleSelect extends SimpleContentType
```
Comment (wachterjohannes) [inline src/Sulu/Bundle/ContentBundle/Content/Types/MultipleSelect.php]: you should provide an empty default parameter for `values`.

File: src/Sulu/Bundle/ContentBundle/Content/Types/SingleSelect.php [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+ */
+
+namespace Sulu\Bundle\ContentBundle\Content\Types;
+
+use Sulu\Component\Content\SimpleContentType;
+
+/**
+ * ContentType for single selects. Currently only supports radios.
+ */
+class SingleSelect extends SimpleContentType
```
Comment (wachterjohannes) [inline src/Sulu/Bundle/ContentBundle/Content/Types/SingleSelect.php]: same here

File: src/Sulu/Bundle/ContentBundle/Content/Types/MultipleSelect.php (hunk 1 above)
Comment (danrot) [inline src/Sulu/Bundle/ContentBundle/Content/Types/MultipleSelect.php]: Where? I am already [providing a default value](https://github.com/sulu-io/sulu/pull/1777/files#diff-5def847f34085bb8efbb8af98c371995R27).

File: src/Sulu/Bundle/ContentBundle/Content/Types/SingleSelect.php (hunk 2 above)
Comment (danrot) [inline src/Sulu/Bundle/ContentBundle/Content/Types/SingleSelect.php]: Same as above: https://github.com/sulu-io/sulu/pull/1777/files#diff-49665dfc303b1e7baa8317246eaca8e0R27

File: src/Sulu/Bundle/ContentBundle/Content/Types/MultipleSelect.php (hunk 1 above)
Comment (wachterjohannes) [inline src/Sulu/Bundle/ContentBundle/Content/Types/MultipleSelect.php]: for the property param with the function `getDefaultParameter` in the content type

File: src/Sulu/Bundle/ContentBundle/Tests/Features/content_type_multiple_select.feature [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+
+    Background:
+        Given there exists a page template "multiple_select_page" with the following property configuration
+        """
+        <property name="select" type="multiple_select">
+            <meta>
+                <title lang="de">Select</title>
+            </meta>
+            <params>
+                <param name="values" type="collection">
```
Comment (wachterjohannes) [inline src/Sulu/Bundle/ContentBundle/Tests/Features/content_type_multiple_select.feature]: if someone forgets to set this property param the generation of form fails. therefor you should provide a default parameter in the content type

Comment (danrot) [inline src/Sulu/Bundle/ContentBundle/Tests/Features/content_type_multiple_select.feature]: good point, fixed that

**Pull request merged**